dicodile/utils/tests/test_csc.py

```diff
@@ -30,3 +33,59 @@ def test_ztz(valid_support, atom_support, sparsity):
     X_hat = reconstruct(z, D)
 
     assert np.isclose(cost, np.dot(X_hat.ravel(), X_hat.ravel()))
+
+
+@pytest.mark.parametrize('atom_support',
+                         [(35, ), (40, 30), (10, 12, 3)])
+def test_dense_convolve_multi_uv_shape(atom_support):
+
+    n_channels = 3
+    sig_shape = (n_channels, *[8 * n for n in atom_support])
+    n_atoms = 25
+    valid_support = get_valid_support(sig_support=sig_shape[1:],
+                                      atom_support=atom_support)
+
+    z_hat = np.ones((n_atoms, *valid_support))
+    u = np.ones((n_atoms, n_channels))
+    v = np.ones((n_atoms, *atom_support))
+    Xi = _dense_convolve_multi_uv(z_hat, (u, v))
+
+    assert Xi.shape == sig_shape
+
+
+def test_convolve_uv_and_convolve_d_match():
+    n_channels = 3
+    sig_shape = (n_channels, 800, 600)
+    atom_shape = (n_channels, 40, 30)
+    atom_support = atom_shape[1:]
+    n_atoms = 20
+    valid_support = get_valid_support(sig_support=sig_shape[1:],
+                                      atom_support=atom_support)
+    rng = np.random.default_rng(seed=42)
+    z_hat = rng.uniform(size=(n_atoms, *valid_support))
+
+    u = rng.uniform(size=(n_atoms, n_channels))
+    v = rng.uniform(size=(n_atoms, *atom_support))
+    uv_convolution = _dense_convolve_multi_uv(z_hat, (u, v))
```

Comment:
To make the coverage work, could you call `_choose_convolve_multi`?

Reply:
makes sense, thanks for the suggestion!